src/app/mod.rs

```diff
@@ -328,84 +327,98 @@ pub async fn run_app(cli: cli::DftCli, state: state::AppState<'_>) -> Result<()>
     app.exit()
 }
 
-pub async fn execute_files_or_commands(
-    files: Vec<PathBuf>,
-    commands: Vec<String>,
-    state: &state::AppState<'_>,
-) -> Result<()> {
-    match (files.is_empty(), commands.is_empty()) {
-        (true, true) => Err(eyre!("No files or commands provided to execute")),
-        (false, true) => execute_files(files, state).await,
-        (true, false) => execute_commands(commands, state).await,
-        (false, false) => Err(eyre!(
-            "Cannot execute both files and commands at the same time"
-        )),
-    }
+/// Encapsulates the command line interface
```

Comment:
I refactored a bunch of free functions into a struct (which I plan to move to its own module soon)